Resources/Public/Javascript/OpenLayers/ImageProcessingUtility.js

```diff
@@ -0,0 +1,225 @@
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+var imageProcessingUtility = {
+
+  'filter': function(canvas, filters) {
+    if (canvas !== undefined) {
+      var ctx = canvas.getContext('2d');
+      for (var filter in filters) {
+        this.applyFilter(canvas, filter, filters[filter]);
+      }
+    }
+  },
+
+  'applyFilter': function(canvas, filter, value) {
+    var ctx = canvas.getContext('2d');
+    var imgData = ctx.getImageData(0, 0, ctx.canvas.width, ctx.canvas.height);
+    switch (filter) {
+      case 'contrast':
+        imgData = this.contrast(imgData, value);
+        ctx.putImageData(imgData, 0, 0);
+        break;
+      case 'hue':
+        imgData = this.hue(imgData, value);
+        ctx.putImageData(imgData, 0, 0);
+        break;
+      case 'saturation':
+        imgData = this.saturation(imgData, value);
+        ctx.putImageData(imgData, 0, 0);
+        break;
+      case 'brightness':
+        imgData = this.brightness(imgData, value);
+        ctx.putImageData(imgData, 0, 0);
+        break;
+      case 'invert':
+        imgData = this.invert(imgData);
+        ctx.putImageData(imgData, 0, 0);
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  'contrast': function (imgData, value) {
+    var data = imgData.data;
+    var contrast = (value * 100 - 100) * 2.55;
+    var factor = (255 + contrast) / (255.01 - contrast);
+
+    for(var i=0;i<data.length;i+=4)
```

Comment:
Why iteration is increased by 4? Is it connected to the alpha? If yes, maybe comment which explains it would be useful. 

Whitespaces are missing.